rosetta-source/src/main/rosetta/product-collateral-func.rosetta

```diff
@@ -1,22 +1,165 @@
     output:
         specification EligibleCollateralSpecification (1..1)
 
     add specification -> criteria:
         instruction -> variable
-            extract MergeEligibleCollateralCriteria(item, instruction->common)
+            extract MergeEligibleCollateralCriteria(item, instruction -> common)
 
 func MergeEligibleCollateralCriteria: <"Java implementation merges criteria1 and criteria2.">
-
     inputs:
         criteria1 EligibleCollateralCriteria (1..1)
         criteria2 EligibleCollateralCriteria (1..1)
     output:
-        criteria EligibleCollateralCriteria (1..1)
\ No newline at end of file
+        criteria EligibleCollateralCriteria (1..1)
+
+
+func CheckEligibility:
+    inputs:
+        specifications EligibleCollateralSpecification (1..*)
+        assetCountry string (1..1)
+        assetType AssetTypeEnum (1..1)
+        securityType SecurityTypeEnum (1..1)
+        issuerType IssuerTypeEnum (1..1)
+        maturity number (1..1)
+    output:
+        eligibilityResult CheckEligibilityResult (0..1)
+
+   alias eligibleSpecifications: specifications 
+       then filter Get_AssetCountryOfOrigin = assetCountry
+       then filter Get_AssetTypeE = assetType
+       then filter Get_IssuerType = issuerType
+       then filter Get_SecurityType = securityType
+       then filter maturity > Get_MaturityLowerBound
+       then filter maturity < Get_MaturityUpperBound
+   
+   add eligibilityResult -> eligibleSummary: eligibleSpecifications then extract Create_ScheduleSummary
+   add eligibilityResult -> specification: eligibleSpecifications 
+   set eligibilityResult -> isEligible: eligibleSpecifications count > 0
+
+// func ReadableEligibility:
+//     inputs:
+//         specification EligibleCollateralSpecification (1..1)
+//     output:
+//         readable string (1..1)
+
+//     alias summary: Create_ScheduleSummary(specification)
+
+//     set readable:  
+//             summary -> assetCountryOfOrigin
+//             + ' ' 
+//             + summary -> assetType
+//             + ' ' 
+//             + // res 
+//             // + ' '
+//             +
+//             summary -> issuerType
+//             + ' ' 
+//             + summary -> haircutPercentage
+
+
+
+func Create_ScheduleSummary:
+    inputs:
+        specification EligibleCollateralSpecification (1..1)
+    output:
+        scheduleSummary ScheduleSummary (1..1)
+
+    set scheduleSummary -> assetCountryOfOrigin: Get_AssetCountryOfOrigin(specification)
+    set scheduleSummary -> assetType: Get_AssetTypeE(specification)
+    set scheduleSummary -> securityType: Get_SecurityType(specification)
+    set scheduleSummary -> maturityLower: Get_MaturityLowerBound(specification)
+    set scheduleSummary -> maturityUpper: Get_MaturityUpperBound(specification)
+    set scheduleSummary -> issuerType: Get_IssuerType(specification)
+    set scheduleSummary -> haircutPercentage: Get_HaircutPercentage(specification)
+
+type CheckEligibilityResult:
```

Comment:
Is the introduction of this type necessary for now ? It is not used anywhere that seems odd